evennia/contrib/utils/name_generator/namegen.py

```diff
@@ -0,0 +1,432 @@
+    _SURNAME_LIST += custom_last_names
+
+
+
+def fantasy_name(num=1, style="harsh", return_list=False):
+    """
+    Generate made-up names in one of a number of "styles".
+
+    Keyword args:
+        num (int)      - How many names to return.
+        style (string) - The "style" of name. This references an existing algorithm.
+        return_list (bool) - Whether to always return a list. `False` by default,
+                which returns a string if there is only one value and a list if more.
+    """
+    # validate num first
+    num = int(num)
+    if num < 1:
+        raise ValueError("Number of names to generate must be positive.")
+
+    if style not in _FANTASY_NAME_STRUCTURES:
+        raise ValueError(f"Invalid style name: '{style}'.")
+    style_dict = _FANTASY_NAME_STRUCTURES[style]
+
+    syllable = []
+    weight = 8
+    # parse out the syllable structure with weights
+    for key in style_dict["syllable"]:
+        # parentheses mean optional - allow nested parens
+        if key == "(":
+            weight = weight/2
+        elif key == ")":
+            weight = weight*2
+        else:
+            if key == "C":
+                type = "consonants"
+            elif key == "V":
+                type = "vowels"
+            else:
+                type = key
+            # append the sound type and weight
+            syllable.append( (type, int(weight)) )
+    
+    name_list = []
+    
+    # time to generate a name!
+    for n in range(num):
+        # build a list of syllables
+        length = random.randint(*style_dict['length'])
+        name = ""
+        for i in range(length):
+            # build the syllable itself
+            syll = ""
+            for sound, weight in syllable:
+                # random chance to skip this key; lower weights mean less likely
+                if random.randint(0,8) > weight:
+                    continue
+                
+                if sound not in style_dict:
+                    # extra character, like apostrophes
+                    syll += sound
+                    continue
+
+                # get a random sound from the sound list
+                choices = list(style_dict[sound])
+
+                if sound == "consonants":
+                    # if it's a starting consonant, add starting-sounds to the options
+                    if not len(syll):
+                        choices += style_dict.get('start',[])
+                    # if it's an ending consonant, add ending-sounds to the options
+                    elif i+1 == length:
+                        choices += style_dict.get('end',[])
+
+                syll += random.choice(choices)
+
+            name += syll
+
+        # condense repeating letters down to a maximum of 2
+        name = _RE_DOUBLES.sub(lambda m: m.group(1)*2, name)
+        # capitalize the first letter
+        name = name[0].upper() + name[1:] if len(name) > 1 else name.upper()
+        name_list.append(name)
```

Comment:
Hmm, good catch. This function is the one that generates made-up names, so names that'd come from name lists like `Tom from Battlebrook` would be unaffected, but it _would_ prevent generating names like `Gelaniel d'Elendal` and `Xaruth x'Thokk`. I'll have to think through the best way to solve it.

Reply:
I think the best route here is for me to write up a docs section on how to use `fantasy_name()` to generate multi-word names which covers the the topic of including custom prefixes into your results, rather than try to broaden the scope of the generator itself.